src/driver/platform_rpi.cpp

```diff
@@ -0,0 +1,140 @@
+
+/****************************************************************
+ * platform_rpi.cpp   
+ * HSCDTD008A Library Extention for RPI
+ * Tilman Glötzner
+ * Original Creation Date: 2022-09-07
+ *
+ * Distributed as-is; no warranty is given.
+ ***************************************************************/
+
+
+#include "platform.h"
+#include "hscdtd008a_driver.h"
+#include <stdio.h>
+#include <sys/ioctl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <unistd.h>
+#include <string.h>
+#include <linux/i2c-dev.h>
+#define I2C_ADDR 0x0f
+#define device "/dev/i2c-1"
+#ifndef I2C_M_RD
+#include <linux/i2c.h>
+#endif
+
+int fd = -1;
+
+int8_t t_open(void)
+{
+    fd = open(device, O_RDWR); 
+    if (fd >= 0)
+    {
+       if (ioctl(fd, I2C_SLAVE, I2C_ADDR) < 0) {
+        printf("ioctl error: %s\n", strerror(errno));
+        return 1;
+       }
+    }   
+    return (fd<0);
+}
+
+int8_t t_read_register(uint8_t addr,
+                       uint8_t reg,
+                       uint8_t length,
+                       uint8_t *p_buffer)
+{
+    int8_t status;
+    uint8_t inbuf[32];
+    uint8_t outbuf[1];
+    struct i2c_msg msgs[2];
+    struct i2c_rdwr_ioctl_data msgset[1];
+
+    msgs[0].addr = addr;
+    msgs[0].flags = 0;
+    msgs[0].len = 1;
+    msgs[0].buf = outbuf;
+ 
+    outbuf[0] = reg;
+    
+    msgs[1].addr = addr;
+    msgs[1].flags = I2C_M_RD | I2C_M_NOSTART;
+    msgs[1].len = length;
+    msgs[1].buf = inbuf;
+
+    msgset[0].msgs = msgs;
+    msgset[0].nmsgs = 2;
+
+
+    if (length >= 32) 
```

Comment:
Why is 32 special? Please explain with a comment.

Reply:
Do you mean start reading at a register and then keep reading? Like for sensor data, that is supported (as well as other registers). If you mean something else, I am not sure what you mean.

Reply:
No, I meant to say reading (or writing) an address range. The HSCDTD008A has an address range from 0x00 - 0x32. I wonder if there is a bulk read that allows the user to read for instance the full address range of 0x00 -- 0x32 with one read operation.

Reply:
With address range, I mean the address range of the registers -- not the I2c address range ...

Reply:
Ah, I don't think this is something for the platform file to validate. The platform file should not need this kind of validation, as this would be shared between all platforms, not just the Raspberry Pi.

It is probably best to remove it from the platform file.

I can add some form of validation check to the transport module instead. Also, as you mentioned, the address range goes from **0x**00 to **0x**32 Hex 0x32 is 50 in decimal.

Reply:
This validation has been added here [eb3646c](https://github.com/bobveringa/HSCDTD008A-Library/pull/10/commits/eb3646c1900653fc4085d1f32cb5281c5ae5fcfc).

Reply:
The buffer is defined and copied in the read/write routine themselves -- meaning that I need to check them in the routine itself for an overflow to avoid misuse.

Reply:
These platform functions are only called by the transport layer, that would trigger the validation. But I understand your position. I don't think it hurts to have this code here.

Marked as resolved